scrapy/utils/signal.py

```diff
@@ -1,79 +1,41 @@
 
 
-class _IgnoredException(Exception):
-    pass
+warnings.warn("`scrapy.utils.signal` is deprecated, signals are now objects of"
+              " the `scrapy.dispatch.Signal` class with the utility functions"
+              " available as instance method of the same.",
+              ScrapyDeprecationWarning, stacklevel=2)
 
 
-def send_catch_log(signal=Any, sender=Anonymous, *arguments, **named):
+def send_catch_log(signal=None, sender=None, *arguments, **named):
     """Like pydispatcher.robust.sendRobust but it also logs errors and returns
     Failures instead of exceptions.
     """
-    dont_log = named.pop('dont_log', _IgnoredException)
-    spider = named.get('spider', None)
-    responses = []
-    for receiver in liveReceivers(getAllReceivers(sender, signal)):
-        try:
-            response = robustApply(receiver, signal=signal, sender=sender,
-                *arguments, **named)
-            if isinstance(response, Deferred):
-                logger.error("Cannot return deferreds from signal handler: %(receiver)s",
-                             {'receiver': receiver}, extra={'spider': spider})
-        except dont_log:
-            result = Failure()
-        except Exception:
-            result = Failure()
-            logger.error("Error caught on signal handler: %(receiver)s",
-                         {'receiver': receiver},
-                         exc_info=True, extra={'spider': spider})
-        else:
-            result = response
-        responses.append((receiver, result))
-    return responses
+    return signal.send_catch_log(sender=sender, **named)
 
 
-def send_catch_log_deferred(signal=Any, sender=Anonymous, *arguments, **named):
+def send_catch_log_deferred(signal=None, sender=None, *arguments, **named):
     """Like send_catch_log but supports returning deferreds on signal handlers.
     Returns a deferred that gets fired once all signal handlers deferreds were
     fired.
     """
-    def logerror(failure, recv):
-        if dont_log is None or not isinstance(failure.value, dont_log):
-            logger.error("Error caught on signal handler: %(receiver)s",
-                         {'receiver': recv},
-                         exc_info=failure_to_exc_info(failure),
-                         extra={'spider': spider})
-        return failure
-
-    dont_log = named.pop('dont_log', None)
-    spider = named.get('spider', None)
-    dfds = []
-    for receiver in liveReceivers(getAllReceivers(sender, signal)):
-        d = maybeDeferred(robustApply, receiver, signal=signal, sender=sender,
-                *arguments, **named)
-        d.addErrback(logerror, receiver)
-        d.addBoth(lambda result: (receiver, result))
-        dfds.append(d)
-    d = DeferredList(dfds)
-    d.addCallback(lambda out: [x[1] for x in out])
-    return d
+    return signal.send_catch_log_deferred(sender=sender, **named)
 
 
-def disconnect_all(signal=Any, sender=Any):
+def disconnect_all(signal, sender=None):
```

Comment:
Previously signal argument was not required; also, it was posisble to pass object instead of Signal instance.